CIME/case/case.py

```diff
@@ -184,12 +184,7 @@ def __init__(self, case_root=None, read_only=True, record=False, non_local=False
                     machobj = Machines(machine=mach, extra_machines_dir=extra_machdir)
                 else:
                     machobj = Machines(machine=mach)
-                probed_machine = machobj.probe_machine_name()
```

Comment:
This check was added because there are instances of multiple supported machines which share a common file system.  Users were creating a case on one and then trying to use the case on another which generated a nonsensical error.   This probe was added to create a reasonable error message.   Perhaps rather than remove this we can discuss why it is causing issues on your systems.

Reply:
The issue associated with this change: https://github.com/ESMCI/cime/issues/4178

Reply:
One case was cori which is one front end for two machines (cori-knl and cori-haswell).   You can't tell which one you're intending to run on by probing the login machine name.   For connivence, the probed name is set to "cori-knl" and if you want to run on haswell you have to use the --machine argument.  This check overrides what the user specified with --machine.

Reply:
Good point - we can fix that by introducing the alternative NODENAME_REGEX as we use on 
perlmutter, that is:  <NODENAME_REGEX>$ENV{NERSC_HOST}:perlmutter</NODENAME_REGEX>
We can add the same for cori-knl.

Reply:
How does that work and where does it get added?

I still think that probe statement should be surrounded by an if (user didn't specify machine).

Reply:
It's in the machine definition in config_machines.xml.   NERSC sets an env variable, NERSC_HOST, for each system.
I tested this at NCAR where we have a common file system but different front ends, nersc has both the filesystem and the frontend in common.    I'll try it at nersc and see if I can refine the solution.

Reply:
I don't think NERSC_HOST env var will help here as it is "cori". We login to cori and then user can build&submit for 2 diff types of compute nodes that are chosen via slurm batch.